xonsh/environ.py

```diff
@@ -454,8 +457,9 @@ def multiline_prompt():
     'TITLE': DEFAULT_TITLE,
     'MULTILINE_PROMPT': '.',
     'XONSHRC': os.path.expanduser('~/.xonshrc'),
-    'XONSH_HISTORY_SIZE': 8128,
-    'XONSH_HISTORY_FILE': os.path.expanduser('~/.xonsh_history'),
+    'XONSH_HISTORY_SIZE': (8128, 'files'),
+    'XONSH_HISTORY_FILE': os.path.expanduser('~/.xonsh_history.json'),
+    'XONSH_STORE_STDOUT': True,
```

Comment:
Good question @adqm.  So I think that it should be on by default. This is because my main motivation for writing it up like this was to enable reproducible workflows. Some things I plan on implementing in near future are a replay functionality as well as a semantic diffing of history files. Not having stdout recording on makes it impossible to recover "OMG what did it just do!" moments.

That said, people will certainly have different limits for what they want to store. This is why I implemented garbage collection based on total file size.  Maybe we should change the default `XONSH_HISTORY_SIZE` from `(8128, 'files')` to be `(16, 'MB')` or some other, similar small number.  

 IMO this is going to be a feature that really sets xonsh apart from other shells.  But it is new.  Maybe we should have a buy-in moment when xonsh is first started. For example, we could have a warning message on startup, until they set a certain config variable.  The user would only have to  do this once and could either turn it on or off, by default, as well as set a default size.  How does that sound?